src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -15,34 +16,88 @@ import {ICoExpressionPlotProps} from "./CoExpressionPlot";
 import {bind} from "bind-decorator";
 import OqlStatusBanner from "../../../shared/components/oqlStatusBanner/OqlStatusBanner";
 import {getMobxPromiseGroupStatus} from "../../../shared/lib/getMobxPromiseGroupStatus";
-import MolecularProfileSelector from "../../../shared/components/MolecularProfileSelector";
+import {IDataQueryFilter} from "shared/lib/StoreUtils";
 
 export interface ICoExpressionTabProps {
     store:ResultsViewPageStore;
 }
 
-export class CoExpressionCache extends MobxPromiseCache<{entrezGeneId:number, molecularProfile:MolecularProfile, allData:boolean}, CoExpression[]> {};
+export class CoExpressionCache extends MobxPromiseCache<{profileX: MolecularProfile, 
+    profileY: MolecularProfile, geneticEntityId: string,
+    geneticEntityType: string, allData:boolean}, CoExpression[]> {}
 
 @observer
 export default class CoExpressionTab extends React.Component<ICoExpressionTabProps, {}> {
-    @observable _selectedMolecularProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
-    @observable _selectedEntrezGeneId:number | undefined; // only undefined initially, until genes downloaded
+    @observable _selectedProfileXGene:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedProfileXGeneset:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
```

Comment:
can you ever select a gene profile and a geneset profile at the same time? if not then these shouldnt be handled as two separate things.

Reply:
but there is only one profile ever selected, so why should you store them separately?